src/v/pandaproxy/schema_registry/subject_name_strategy.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "pandaproxy/schema_registry/types.h"
+#include "utils/string_switch.h"
+
+namespace pandaproxy::schema_registry {
+
+enum class subject_name_strategy : uint8_t {
+    topic_name,
+    record_name,
+    topic_record_name,
+};
+
+inline constexpr std::string_view to_string_view(subject_name_strategy e) {
+    switch (e) {
+    case subject_name_strategy::topic_name:
+        return "io.confluent.kafka.serializers.subject.TopicNameStrategy";
+    case subject_name_strategy::record_name:
+        return "io.confluent.kafka.serializers.subject.RecordNameStrategy";
+    case subject_name_strategy::topic_record_name:
+        return "io.confluent.kafka.serializers.subject.TopicRecordNameStrategy";
+    }
+    return "{invalid}";
+}
```

Comment:
>would love to not use these names, but I understand why :)

do we need to use the `invalid` return value? if the switch is complete over the enum values then it can be ommitted  huh?.

Reply:
right, see comment above on `redpanda` + `confluent.*` for compat, but we should print `redpanda.*` here.

Reply:
also, looking at the prefix, maybe we drop all the hungarian notation and just use the last dot.string so `TopicRecordNameStrategy, RecordNameStrategy, TopicNameStrategy` ... they all seem unique to me, no need to print prefix?

Reply:
Habits